lib/redis_memo/memoize_method.rb

```diff
@@ -45,19 +45,15 @@ def memoize_method(method_name, method_id: nil, **options, &depends_on)
     @memoized_dependencies[method_name] = depends_on
 
     define_method :dependency_of do |method_name, *method_args|
-      self.class.dependency_of(method_name, *method_args)
-    end
-  end
-
-  def dependency_of(method_name, *method_args)
-    depends_on = @memoized_dependencies[method_name]
-    unless depends_on
-      raise(
-        RedisMemo::ArgumentError,
-        "#{method_name} is not a memoized method"
-      )
+      method_depends_on = self.class.instance_variable_get(:@memoized_dependencies)[method_name]
```

Comment:
maybe we should prefix this instance var with redis_memo to avoid potential conflicts?